k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/resolver/CredentialsResourceHandler.kt

```diff
@@ -82,16 +87,16 @@ class CredentialsResourceHandler(
             SECRET,
             mapOf(
                 "namespace" to resource.spec.namespace,
-                "name" to "${resource.spec.template.metadata["type"]}-${resource.spec.template.data?.account}",
+                "name" to "${resource.spec.template.metadata["type"]}-${clouddriverAccountName}",
                 "annotations" to mapOf("strategy.spinnaker.io/versioned" to "false")
             ),
             null,
-            data
+            data.toMap() as Map<String, Any?>?
```

Comment:
if we remove `K8sData` this can be a simple map too so no converting back and forth.